File: LeetCode/wordbreak.java
public class Solution {
    public boolean wordBreak(String s, Set<String> wordDict) {
        int[] pos = new int[s.length()+1];
        Arrays.fill(pos,-1);
        pos[0] = 0;
        for(int i=0;i<s.length();i++)
        {
            if(pos[i]==-1)
                continue;
            for(int j=i+1;j<=s.length();j++)
            {
                String subs = s.substring(i,j);
                if(wordDict.contains(subs))
                    pos[j] = i;
            }
        }
        
        return pos[s.length()]!=-1;
    }
}

File: LeetCode/pythonSols/DynamicProg/wordBreak.py
class Solution(object):
    def wordBreak(self, s, wordDict):
        """
        :type s: str
        :type wordDict: Set[str]
        :rtype: bool
        """
        slength = len(s)
        dp = [False]*(slength+1)
        dp[0] = True
        for i in range(1,slength+1):
            for j in range(i):
                if dp[j] and s[j:i] in wordDict:
                    dp[i] = True
                    break
        return dp[slength]